jax/experimental/stax.py

```diff
@@ -339,6 +342,81 @@ def apply_fun(params, inputs, **kwargs):
     return init_fun, apply_fun
 
 
+def LSTM(out_dim, W_init=glorot(), b_init=normal()):
+    """Layer construction function for a Long-Short Term Memory (LSTM) layer."""
+    def init_fun(rng, input_shape):
+        k1, k2 = random.split(rng)
+        cell, hidden = (b_init(k1, (input_shape[0], out_dim)),
+                        b_init(k2, (input_shape[0], out_dim)))
+
+        k1, k2, k3 = random.split(k1, num=3)
+        forget_W, forget_U, forget_b = (
+            W_init(k1, (input_shape[2], out_dim)),
+            W_init(k2, (out_dim, out_dim)),
+            b_init(k3, (out_dim,)),
+        )
+
+        k1, k2, k3 = random.split(k1, num=3)
+        in_W, in_U, in_b = (
+            W_init(k1, (input_shape[2], out_dim)),
+            W_init(k2, (out_dim, out_dim)),
+            b_init(k3, (out_dim,)),
+        )
+
+
+        k1, k2, k3 = random.split(k1, num=3)
+        out_W, out_U, out_b = (
+            W_init(k1, (input_shape[2], out_dim)),
+            W_init(k2, (out_dim, out_dim)),
+            b_init(k3, (out_dim,)),
+        )
+
+        k1, k2, k3 = random.split(k1, num=3)
+        change_W, change_U, change_b = (
+            W_init(k1, (input_shape[2], out_dim)),
+            W_init(k2, (out_dim, out_dim)),
+            b_init(k3, (out_dim,)),
+        )
+
+        output_shape = (input_shape[0], out_dim)
```

Comment:
Should this not be output_shape = (input_shape[0], input_shape[1], out_dim)? If I understand correctly the input tensor is (batches,sequences,features) so the output tensor is (batches,sequences,out_dim)?

Reply:
Thanks David for taking the time! Yes, you are right - I will change that asap.

Reply:
I noticed the other stax layers do this

output_shape = input_shape[:-1] + (out_dim,)

At first I assumed it just allowed you to type less - but I've realised it's actually allows you to pass an input shape of either (sequences,features) or (batches,sequences,features) - i.e. it gives you a bit more flexibility to train on one or multiple batches which is a nice feature